watchman/rust/watchman_client/src/pdu.rs

```diff
@@ -559,6 +560,121 @@ pub struct SubscribeResponse {
     pub saved_state_info: Option<Value>,
 }
 
+/// The `trigger` command request.
+///
+/// The fields are explained in detail here:
+/// <https://facebook.github.io/watchman/docs/cmd/trigger#extended-syntax>
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct TriggerRequest {
+    /// Defines the name of the trigger.
+    pub name: String,
+
+    /// Specifies the command to invoke.
+    pub command: Vec<String>,
+
+    /// It true, matching files (up to system limits) will be added to the
+    /// command's execution args.
+    #[serde(default, skip_serializing_if = "is_false")]
+    pub append_files: bool,
+
+    /// Specifies the expression used to filter candidate matches.
+    #[serde(skip_serializing_if = "Option::is_none", skip_deserializing)]
+    pub expression: Option<Expr>,
+
+    /// Configure the way `stdin` is configured for the executed trigger.
+    #[serde(
+        default,
+        skip_serializing_if = "TriggerStdinConfig::is_devnull",
+        serialize_with = "TriggerStdinConfig::serialize",
+        skip_deserializing
+    )]
+    pub stdin: TriggerStdinConfig,
```

Comment:
```suggestion
    pub stdin: Option<TriggerStdinConfig>,
```

nit, it looks like you have it set up correctly to default to `/dev/null`, but Watchman knows how to default itself to `/dev/null` if this is omitted, so it probably is best to let Watchman handle this being omitted rather than defaulting it in the client api (in case Watchman changes the behavior of this being omitted): 

https://github.com/facebook/watchman/blob/f6ee0299b7c6dc15ce694d2ead92fae9bce96a22/watchman/TriggerCommand.cpp#L349-L352

Reply:
The use of `skip_serializing_if = "TriggerStdinConfig::is_devnull",` on line 587 accomplishes the same thing but means that I don't have to deal with an extra layer of wrap/unwrap in all the surrounding code

Reply:
oh, I understand what you mean.  Will change it.